### PR TITLE
Browse safely after erasure

### PR DESCRIPTION
**Pull request by hrhino**
Kinda like those virus-scanning toolbars used to advertise.

Probably no one but me and Dale (who asked for it lo so many years back) will notice this, but let's do it anyways. The opportunity for a "why browse?" pun's been taken away from me, but the feature does get some use.

I think this is the only missing type that could plausibly be on a tree at the end of a phase.

Fixes scala/bug#9774.

Also have a "go to unit" feature, for when you've got a lot of files.

### CODE REVIEW

**Review by dwijnand — APPROVED** [review]
Thanks!  I can barely remember running into this while trying to fix value classes in the compiler :D

**Review by retronym — APPROVED** [review]

**Comment by som-snytt** [comment]
Yowsa browsa!

File: src/compiler/scala/tools/nsc/ast/TreeBrowsers.scala [hunk 1]
```diff
@@ -279,6 +279,29 @@ abstract class TreeBrowsers {
         }
       )
       jmView add jmiCollapse
+      val jmiGoto = new JMenuItem(
+        new AbstractAction("Go to unit") {
+          putValue(Action.ACCELERATOR_KEY, KeyStroke.getKeyStroke(KeyEvent.VK_N, menuKey, false))
+          override def actionPerformed(actionEvent: ActionEvent) {
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/ast/TreeBrowsers.scala:285]: If linting were on, we could better appreciate the pun on "go to unit:" within procedure syntax that omits `:Unit`.

Comment (hrhino) [reply src/compiler/scala/tools/nsc/ast/TreeBrowsers.scala:285]: It's a last hurrah, of sorts. I assume that this PR means that @dwijnand can finally fix value classes for good.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/ast/TreeBrowsers.scala:285]: That's better than fixing them for worse.

**Pull request merged**